src/components/screens/Consul/Proposals/NewProposal.js

```diff
@@ -0,0 +1,361 @@
+
+            {item.type === 'title' && (
+              <>
+                <Label>{item.title}</Label>
+              </>
+            )}
+
+            {item.type === 'infoText' && (
+              <>
+                <RegularText smallest placeholder>
+                  {item.title}
+                </RegularText>
+              </>
+            )}
+
+            {item.type === 'category' && (
+              <>
+                <Label>{item.title}</Label>
+                <ScrollView horizontal showsHorizontalScrollIndicator={false}>
+                  {item.category.map((items, indexs) => (
+                    <TouchableOpacity
+                      key={indexs}
+                      style={[
+                        styles.tagContainer,
+                        {
+                          backgroundColor: items.selected ? colors.darkerPrimary : colors.borderRgba
+                        }
+                      ]}
+                      onPress={() => {
+                        if (items.selected) {
+                          items.selected = false;
+                        } else {
+                          items.selected = true;
+                        }
+                        setTags([]);
+                      }}
+                    >
+                      <RegularText
+                        small
+                        style={styles.tagText}
+                        lighter={items.selected ? true : false}
+                      >
+                        {items.name}
+                      </RegularText>
+                    </TouchableOpacity>
+                  ))}
+                </ScrollView>
+              </>
+            )}
+          </Wrapper>
+        ))}
+
+        <WrapperHorizontal>
+          <Checkbox
+            title={text.termsOfServiceLabel}
+            link={'https://beteiligung.bad-belzig.de/conditions'}
+            linkDescription={text.termsOfServiceLinkLabel}
+            checkedIcon="check-square-o"
+            uncheckedIcon="square-o"
+            checked={termsOfService}
+            onPress={() => settermsOfService(!termsOfService)}
+          />
+        </WrapperHorizontal>
+
+        <Wrapper>
+          <Button
+            onPress={handleSubmit(onSubmit)}
+            title={
+              query === queryTypes.START_PROPOSAL
+                ? text.newProposalStartButtonLabel
+                : text.updateButtonLabel
+            }
+          />
+        </Wrapper>
+      </WrapperWithOrientation>
+    </SafeAreaViewFlex>
+  );
+};
+
+NewProposal.propTypes = {
+  navigation: PropTypes.shape({
+    navigate: PropTypes.func.isRequired,
+    goBack: PropTypes.func
+  }).isRequired,
+  data: PropTypes.object,
+  query: PropTypes.string
+};
+
+const styles = StyleSheet.create({
+  tagContainer: {
+    backgroundColor: colors.borderRgba,
+    margin: 5,
+    borderRadius: 5
+  },
+  tagText: {
+    padding: 10
+  },
+  noPaddingTop: {
+    paddingTop: 0
+  }
+});
+
+const Inputs = [
```

Comment:
is there a specific structure which these inputs follow?

Reply:
What kind of follow-up are you talking about?

Reply:
It's a method I've used before. I will investigate this further and create a document.